api/controllers/link-controller.ts

```diff
@@ -29,7 +34,19 @@ export const createLink = async (
     })
     .toArray();
 
-  if (result.length !== 0) return createLink(longUrl, ipAddress);
+  if (result.length !== 0) {
+    if (!customCode) return createLink(longUrl, ipAddress);
+    else {
+       result = await database
+        .find(
+          { shortCode: shortCode }
+        )
+        .toArray();
+    }
+
+    if (result.length !== 0) throw CUSTOM_CODE_ALREADY_EXISTS;
+    else return createLink(longUrl, ipAddress, customCode);
```

Comment:
No need to use if and else with return. Return will stop further execution anyway.